### PR TITLE
Find libnvidia-ml in driver root

### PR DESCRIPTION
**Pull request by elezar**

### CODE REVIEW

File: pkg/nvcdi/lib.go [hunk 1]
```diff
@@ -113,7 +113,16 @@ func New(opts ...Option) (Interface, error) {
 		lib = (*managementlib)(l)
 	case ModeNvml:
 		if l.nvmllib == nil {
-			l.nvmllib = nvml.New()
+			var nvmlOpts []nvml.Option
+			candidates, err := l.driver.Libraries().Locate("libnvidia-ml.so.1")
+			if err != nil {
+				l.logger.Warningf("Ignoring error in locating libnvidia-ml.so.1: %v", err)
```
Comment (cdesiniotis) [inline pkg/nvcdi/lib.go:119]: Why ignore the error? If we can't find `libnvidia-ml.so.1` here then wouldn't we be failing at a later point anyways when we try to initialize NVML?

**Review by cdesiniotis — COMMENTED** [review]
What is the context / motivation for this change?

File: pkg/nvcdi/lib.go (hunk 1 above)
Comment (elezar) [reply pkg/nvcdi/lib.go:119]: The primary reason is that I didn't want to change existing behaviour unintentionally.

**Comment by elezar** [comment]
> What is the context / motivation for this change?

Without this change CDI spec generation fails when running with a driver container.

**Pull request merged**